sklearn/utils/_param_validation.py

```diff
@@ -154,6 +155,10 @@ def validate_params(parameter_constraints):
         Note that the *args and **kwargs parameters are not validated and must not be
         present in the parameter_constraints dictionary.
 
+    prefer_skip_nested_validation : bool, default=True
```

Comment:
That was my first naming, but the ``prefer`` prefix was suggested here https://github.com/scikit-learn/scikit-learn/pull/25815#discussion_r1156619634.

As I don't have a strong opinion on the naming, I used the suggested one even though I agree it's not really a preference but more a flag. I let you and @thomasjpfan decide 😄

Reply:
Concretely, here is a table that maps from global config & parameter to what I think should be the expected behavior:

| global_skip_validation | prefer_skip_nested_validation | skip nested validation? |
|------------------------|-------------------------------|-------------------------|
| True                   | True                          | True                    |
| True                   | False                         | True                    |
| False                  | True                          | True                    |
| False                  | False                         | False                   |

Reply:
Ah. I hadn't even thought this far yet. But it shows that your expectation and mine are different. I had assumed (and not realised it isn't true) that the context manager always "wins". That means if globally `skip_validation=True` and we enter a context maneger that sets `skip_validation=False` then validation isn't skipped.

Did you discuss why it makes sense to implement it the way you did? If we stick with it, I think we should add a comment to the context manager to make it clear that all the context manager can do is enable skipping. It can't overrule the global setting.